cmd/microcloud/ask.go

```diff
@@ -19,12 +19,48 @@ import (
 	"github.com/canonical/lxd/shared/validate"
 	cephTypes "github.com/canonical/microceph/microceph/api/types"
 
+	cloudAPI "github.com/canonical/microcloud/microcloud/api"
 	"github.com/canonical/microcloud/microcloud/api/types"
 	cloudClient "github.com/canonical/microcloud/microcloud/client"
 	"github.com/canonical/microcloud/microcloud/multicast"
 	"github.com/canonical/microcloud/microcloud/service"
 )
 
+func checkInitialized(stateDir string, expectInitialized bool) error {
+	cfg := initConfig{autoSetup: true}
+
+	installedServices := []types.ServiceType{types.MicroCloud, types.LXD}
+	optionalServices := map[types.ServiceType]string{
+		types.MicroCeph: cloudAPI.MicroCephDir,
+		types.MicroOVN:  cloudAPI.MicroOVNDir,
+	}
+
+	installedServices, err := cfg.askMissingServices(installedServices, optionalServices)
+	if err != nil {
+		return err
+	}
+
+	s, err := service.NewHandler("", "", stateDir, installedServices...)
+	if err != nil {
+		return err
+	}
+
+	return s.RunConcurrent("", "", func(s service.Service) error {
```

Comment:
Worth returning an combined error from `RunConcurrent()` in case multiple services are (not) initialized? I think right now it only shows the first error which is encountered.

Reply:
That might be too noisy, the errors will be listed hierarchically due to error wrapping, but they won't be related to each other.